components/core/tests/test-Grep.cpp

```
@@ -31,222 +43,531 @@
+        auto extended_search_string_view2
+                = search_string_view2.extend_to_adjacent_greedy_wildcards();
+        REQUIRE(extended_search_string_view2.surrounded_by_delims_or_wildcards(lexer));
+        REQUIRE(6 == extended_search_string_view2.length());
+        REQUIRE(extended_search_string_view2.get_value() == "test\\*");
+    }
+
+    SECTION("getters") {
+        auto search_string_view = WildcardExpressionView{search_string, 2, search_string.length()};
+        REQUIRE(false == search_string_view.is_greedy_wildcard());
+        REQUIRE(false == search_string_view.is_non_greedy_wildcard());
+        REQUIRE('t' == search_string_view.get_char(0));
```

cppcheck

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)